prow/config/config.go

```diff
@@ -898,16 +927,26 @@ func (c *Config) finalizeJobConfig() error {
 			return errors.New("no default GCS credentials secret provided for plank")
 		}
 
+		if c.Plank.DefaultDecorationConfig != nil && len(c.Plank.DefaultDecorationConfigs) == 0 {
+			logrus.Warning("default_decoration_config will be deprecated in the future and it will be replaced with default_decoration_configs['*'].")
+		}
+
+		if len(c.Plank.DefaultDecorationConfigs) > 0 {
+			if _, ok := c.Plank.DefaultDecorationConfigs["*"]; !ok {
```

Comment:
You should populate this from `default_decoration_config` if it is not set

Reply:
(and then the apply logic is simpler, too)

Reply:
addressed